sematic/ui/src/pipelines/external_resource/ExternalResource.tsx

```diff
@@ -0,0 +1,61 @@
+import Timeline from '@mui/lab/Timeline';
+import Alert from "@mui/material/Alert";
+import { useMemo } from "react";
+import { useExternalResource } from "../../hooks/externalResourceHooks";
+import { usePipelinePanelsContext } from "../../hooks/pipelineHooks";
+import { styled } from '@mui/system';
+import ExternalResourceState from './ExternalResourceState';
+import timelineItemClasses from '@mui/lab/TimelineItem/timelineItemClasses';
+import { ExternalResourceHistorySerialization } from '../../Models';
+import { useRunPanelLoadingIndicator } from '../../hooks/runDetailsHooks';
+
+const ThinTimetime = styled(Timeline)`
+    margin: 0;
+    flex: 0;
+    & .${timelineItemClasses.root}:before {
+        flex: 0;
+        padding: 0;
+    }
+`;
+
+export default function ExternalResourcePanel() {
+
+    const { selectedRun } = usePipelinePanelsContext();
+
+    const { value: externalResources, loading, error } = useExternalResource(selectedRun!);
+
+    const historyRecords = useMemo<Array<ExternalResourceHistorySerialization> | null>(
+        () => {
+            if (!externalResources) {
+                return null;
+            }
+            if (externalResources.length === 0) {
+                return [];
+            }
+            const externalResource = externalResources[0];
+            const history 
+                = externalResource.history_serializations as 
+                Array<ExternalResourceHistorySerialization> || [];
+            return history.reverse();
+        }, [externalResources]);
+
+    useRunPanelLoadingIndicator(loading);
+
+    if (!externalResources || externalResources.length === 0) {
+        return <Alert severity="info" sx={{ mt: 3 }}>
+            The run does not use any external resource or the external resource state has not been reported yet.
+        </Alert>
+    }
+
+    if (!!error) {
+        return <Alert severity="error" sx={{ mt: 3 }}> {error.message} </Alert>
+    }
+
+    return <ThinTimetime key={selectedRun?.id}>
+            {historyRecords?.map(
+                (state, index) => 
+                <ExternalResourceState 
+                    historyRecord={state} key={index} isLast={historyRecords.length - 1 === index}/>
+            )}
+        </ThinTimetime>
```

Comment:
Can we display *something* if there is more than one external resource, even if it's only a message that says "The run has additional external resources that are not displayed" or something?

Reply:
Sure, it is going to look like this in case it happens:

<img width="739" alt="image" src="https://user-images.githubusercontent.com/1046489/215232481-b8cf5fbc-99d4-4385-bab0-3a33825f3fd5.png">

Reply:
We can change the wording if needed